src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -267,16 +267,73 @@ Copyright (c) .NET Foundation. All rights reserved.
     <AdditionalProbingPath Include="$(UserProfileRuntimeStorePath)$([System.IO.Path]::DirectorySeparatorChar)|arch|$([System.IO.Path]::DirectorySeparatorChar)|tfm|" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <CompileDependsOn>
+      $(CompileDependsOn)
+      _CreateAppHost;
+    </CompileDependsOn>
+  </PropertyGroup>
+
+  <!--
+    ============================================================
+                                        _CreateAppHost
+    If we found a restored apphost, create the modified destiniation apphost
+    with options from the project.
+    ============================================================
+     -->
+  <UsingTask TaskName="CreateAppHost" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_CreateAppHost"
+          DependsOnTargets="_GetAppHostPaths;CoreCompile"
+          Condition="'$(ComputeNETCoreBuildOutputFiles)' == 'true'">
+    <PropertyGroup>
+      <UseWindowsGraphicalUserInterface Condition="$(RuntimeIdentifier.StartsWith('win')) and '$(OutputType)'=='WinExe'">true</UseWindowsGraphicalUserInterface>
+    </PropertyGroup>
+    <CreateAppHost AppHostSourcePath="@(NativeRestoredAppHostNETCore)"
+                   AppHostDestinationPath="@(NativeAppHostNETCore)"
+                   AppBinaryName="$(AssemblyName)$(TargetExt)"
+                   IntermediateAssembly="@(IntermediateAssembly->'%(FullPath)')"
+                   WindowsGraphicalUserInterface="$(UseWindowsGraphicalUserInterface)"
+                   Condition="'@(NativeRestoredAppHostNETCore)' != '' " />
+  </Target>
+
+  <!--
+    ============================================================
+                                        _GetAppHostPaths
+
+    Gets the path to the restored apphost, and if the restored apphost
+    was present, computes the path for the destination apphost.
+    ============================================================
+     -->
+  <Target Name="_GetAppHostPaths"
+          DependsOnTargets="ResolvePackageAssets">
+    <ItemGroup>
+      <NativeRestoredAppHostNETCore Include="@(NativeCopyLocalItems)"
+                                    Condition="'%(NativeCopyLocalItems.FileName)%(NativeCopyLocalItems.Extension)' == '$(_DotNetAppHostExecutableName)'"/>
+    </ItemGroup>
+
+    <PropertyGroup Condition="'@(NativeRestoredAppHostNETCore)' != '' ">
+      <AppHostDestinationDirectoryPath>$(BaseIntermediateOutputPath)\$(TargetFramework)\$(RuntimeIdentifier)\host</AppHostDestinationDirectoryPath>
```

Comment:
I didn't change the destination apphost path at all in this PR, but I can fix it with your suggestions. Why `$(IntermediateOutputPath)` instead of `$(BaseIntermediateOutputPath)`?
I don't know if there's a reason we put it in the host subfolder. Should I remove that?

Reply:
I see. I'm just reading this for the first time due to the diff. :)

We can change this separately.

Reply:
Ok! I'm happy to do it here as well since it's small, just wanted to understand your suggestions.

Reply:
cc @peterhuene

Reply:
+1 on not needing the host subdirectory.  I can correct that later.

Reply:
Thanks for the explanation. Changed to `IntermediateOutputPath`.